back_end/src/models/requirements.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from typing import List
+
+from ..setup import db
+
+class Requirements(db.Model):
+    tableName = "Requirements"
+
+    id = db.Column(db.Integer, primary_key=True)
+    major_id = db.Column(db.Integer, db.ForeignKey('majors.id'), nullable=True)
+    minor_id = db.Column(db.Integer, db.ForeignKey('minors.id'), nullable=True)
+    class_id = db.Column(db.String, nullable=False)
+    category = db.Column(db.String, nullable=True)
+    subcategory = db.Column(db.String, nullable=True)
+
+    def __init__(self, **kwargs):
+        super(Requirements, self).__init__(**kwargs)
+
+    def to_json(self):
+        ret = {}
+        ret['id'] = self.id
+        ret['major_id'] = self.major_id
+        ret['minor_id'] = self.minor_id
+        ret['class_id'] = self.class_id
+        ret['category'] = self.category
+        ret['subcategory'] = self.subcategory
+
+    def update_attr(self, major_id: int, minor_id: int, class_id: string, 
+                    category: string, subcategory: string) -> bool:
+        if major_id:
+            self.major_id = major_id
+        if minor_id:
+            self.minor_id = minor_id
+        if class_id:
+            self.class_id = class_id
+        if category:
+            self.category = category
+        if subcategory:
+            self.subcategory = subcategory
+        return True
+
+    def save(self):
+        db.session.commit()
+
+    @staticmethod
+    def create_requirement(major_id: int, minor_id: int, class_id: string, 
+                    category: string, subcategory: string) -> bool:
```

Comment:
If you enable flake8, you should see some indentation error. (but it is just for styling, doesn't really matter)